web_external/templates/body/userSubmissionsPage.jade

```diff
@@ -0,0 +1,90 @@
+- var hasSubmissions = submissions.length > 0
+- var hasSuccessfulSubmission = false
+- var endOfSubmissions = !submissions.hasNextPage()
+
+- function getShortLog(job, numLines) {
+-   var logLines = job.get('log').split('\n')
+-   var shortLog = logLines.slice(0, numLines)
+-   if (logLines.length > numLines && numLines > 0) {
+-     shortLog.push('...')
+-   }
+-   return shortLog.join('\n')
+-  }
```

Comment:
I vote we move this function definition out of the jade template and pass it into the template as a variable.